packages/cli/src/lib/file.ts

```diff
@@ -157,3 +158,19 @@ export async function mergePackageJson(
     options?.onResult?.(targetPkgJson) ?? targetPkgJson,
   );
 }
+
+export function mergeTsConfig(sourceDir: string, targetDir: string) {
+  const sourceConfig = ts.readConfigFile(
+    joinPath(sourceDir, 'tsconfig.json'),
+    ts.sys.readFile,
+  ).config;
+
+  if (sourceConfig?.compilerOptions?.types) {
+    replaceFileContent(joinPath(targetDir, 'tsconfig.json'), false, (content) =>
+      content.replace(
+        /"types":\s*\[[^\]]*\]/,
+        `"types": ${JSON.stringify(sourceConfig.compilerOptions.types)}`,
+      ),
+    );
+  }
+}
```

Comment:
@frandiox finally got around to this. Let me know if this implementation is ok

Reply:
Please avoid importing `'typescript'` since that means we would start bundling it and is quite big.
That's why I suggested using regexp to `match` it from the file 🙃  -- you can simply use `readFile` to get its content. It's a file we control and know the contents of, so it's ok to assume this works.